### ISSUE TITLE
bug(github): Inconsistent results based on org order 

### ISSUE THREAD

**Issue by mvbattista** [cloudquery/cloudquery#13163]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

The actual total of repositories of organizations A-D (in the same Github enterprise) are:

- Organization A: 15
- Organization B: 103
- Organization C: 122
- Organization D: 128

Count of repos when orgs are in the following order in `spec.spec.orgs` for multiple clean runs

- A: 15
- C: 3
- B: 122
- D: 28

===

- B: 103
- C: 22
- D: 28
- A: 0

===

- C: 122
- A: 0
- B: 3
- D: 28

===

- D: 128
- A: 0
- C: 22
- B: 3

When each organization is isolated or in the first position will have the complete list.

### Expected Behavior

No matter what the order in `spec.spec.orgs`, all repositories should be pulled into the destination. We plan to configure this for many more organizations, but this bug is a major blocker.

### CloudQuery (redacted) config

```
kind: source
spec:
  destinations:
  - postgresql
  name: github
  path: cloudquery/github
  skip_tables:
  - github_releases
  - github_release_assets
  - github_repository_dependabot_alerts
  - github_repository_dependabot_secrets
  - github_organization_dependabot_alerts
  - github_organization_dependabot_secrets
  spec:
    app_auth:
    - org: <organization_A>
      private_key_path: cloudquery-github-app-key
      app_id: '<app_id>'
      installation_id: '<org_A_installation_id>'
    - org: <organization_B>
      private_key_path: cloudquery-github-app-key
      app_id: '<app_id>'
      installation_id: '<org_B_installation_id>'
    - org: <organization_C>
      private_key_path: cloudquery-github-app-key
      app_id: '<app_id>'
      installation_id: '<org_C_installation_id>'
    - org: <organization_D>
      private_key_path: cloudquery-github-app-key
      app_id: '<app_id>'
      installation_id: '<org_D_installation_id>'
    orgs:
    - <organization_A>
    - <organization_B>
    - <organization_C>
    - <organization_D>
  tables:
  - github_repositories
  - github_organization_members
  - github_organizations
  version: v7.1.3
---
kind: destination
spec:
  name: postgresql
  path: cloudquery/postgresql
  registry: github
  spec:
    connection_string: <connection_string>
  version: v5.0.0
```

### Steps To Reproduce

1. Install github app into multiple github organizations with all read-only permissions
2. Adjust config to account for app installations
3. Run `cloudquery sync`
4. Run destination query such as `SELECT org, COUNT(*) FROM github_repositories GROUP BY org` to view totals
5. Record results and clear destination/point to new destination
6. Rearrange order of orgs in `spec.spec.orgs`
7. Run `cloudquery sync`
8. Run destination query such as `SELECT org, COUNT(*) FROM github_repositories GROUP BY org` to view totals

### CloudQuery (redacted) logs

When run for order:
    - B
    - C
    - D
    - A
[cloudquery_bug_cleaned.log](https://github.com/cloudquery/cloudquery/files/12364616/cloudquery_bug_cleaned.log)

### CloudQuery version

cloudquery version 3.13.1

### Additional Context

Willing to provide more content.

### Pull request (optional)

- [ ] I can submit a pull request

**Comment by erezrokah** [comment]
Thanks for opening @mvbattista the issue, I'll take a look

**Comment by erezrokah** [comment]
Hi @mvbattista 👋 Can you try with version [v7.1.4](https://github.com/cloudquery/cloudquery/releases/tag/plugins-source-github-v7.1.4)? It should have a fix for this issue.

On another note, you can use `skip_dependent_tables: true` to skip child tables without specifying each one. See https://www.cloudquery.io/docs/reference/source-spec#skip_dependent_tables

**Comment by erezrokah** [comment]
I also opened https://github.com/cloudquery/cloudquery/pull/13188 so users can speed up the discovery process with the possible tradeoff of hitting rate limits

**Comment by mvbattista** [comment]
LGTM. Repos count match, and going to look at discovery concurrency usage for our organizations.

**Comment by erezrokah** [comment]
Thanks for following up. Discovery concurrency is released in [`v7.2.0`](https://github.com/cloudquery/cloudquery/releases/tag/plugins-source-github-v7.2.0)
